internal/msalbase/token_response.go

```diff
@@ -114,11 +116,12 @@ func CreateTokenResponse(authParameters *AuthParametersInternal, responseCode in
 
 	idToken, err := CreateIDToken(payload.IDToken)
 	if err != nil {
-		//ID tokens aren't always returned, so the error is just logged
+		// ID tokens aren't always returned, so the error is just logged
+		// TODO(jdoak): we should probably remove this. Either this is an error or isn't.
 		log.Errorf("ID Token error: %v", err)
```

Comment:
please remove

Reply:
I'm unclear on the instruction.  So the original code will return a TokenResponse{} regardless of getting an IDToken.  
Which should I do:
- Change CreateIDToken() to never error?
- Ignore the error here and put in a comment that says why (which I'm not sure, so you'd have to fill in)
- return this error, stop processing

Reply:
IdTokens aren't always returned, for example in service 2 service / client_credential grant where there is no user (and so no idToken). This is not an error. `log.Info` or `log.Verbose` is more suitable.

Reply:
You do not need the IdToken to create an `AuthenticationResult` but I think as long as there is a ClientInfo, you can rely on there being an IdToken as well. 

In fact, you should be able to create AuthenticationResult from **either** Access Token **or** Id Token, i.e. one of them can be missing. Here's the algorithm from MSAL.NET https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/01203edda53f4850b6a02f3569127f68086f5b8c/src/client/Microsoft.Identity.Client/AuthenticationResult.cs#L117

Reply:
Not saying you have to improve this area of MSAL.go - just giving some context. I'm aware that this PR is too large already, my goal is to bring MSAL context and help close on remaining issues.